llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2804,12 +2728,10 @@ void VPlanTransforms::convertToConcreteRecipes(VPlan &Plan,
       ToRemove.push_back(VPI);
     }
     for (VPRecipeBase &R : make_early_inc_range(*VPBB)) {
-      if (auto *ExtRed = dyn_cast<VPExtendedReductionRecipe>(&R)) {
-        expandVPExtendedReduction(ExtRed);
-        continue;
+      if (auto *Bundle = dyn_cast<VPSingleDefBundleRecipe>(&R)) {
+        Bundle->unbundle();
+        Bundle->eraseFromParent();
       }
```

Comment:
Should this unbundling take place during the above loop over VPBB's recipes, or before it?
Should `Bundle` be added to `ToRemove`?

Reply:
Yep, updated, thanks